### PR TITLE
Add check to ensure loaded port matches the declared version

### PR DESCRIPTION
**Pull request by ras0219-msft**
Fixes https://github.com/microsoft/vcpkg/issues/20503.

The fundamental issue in the thread was that the user's versions database did not match the actual retrieved files. This PR addresses the problem by adding an integrity check to ensure that files retrieved from a registry match the version declared in that registry's database files.

Before:
```
❯ vcpkg install --triplet x64-windows
Error: Cycle detected during arrow:x64-windows:
bloom-filter:x64-windows
```
After:
```
❯ vcpkg install --triplet x64-windows
Error: Failed to load port from D:\src\cycle_detected\./vcpkg_registry\ports/arrow: version specs did not match: 'arrow@6.0.0.20210925#4' != 'arrow@6.0.0.20210925'
Error: Failed to load port from D:\src\cycle_detected\./vcpkg_registry\ports/arrow: version specs did not match: 'arrow@6.0.0.20210925#4' != 'arrow@6.0.0.20210925'
```

The error is emitted twice because it is checked every time the algorithm attempts to load the file. The errors are accumulated and presented at once, which currently results in duplication.

Potential improvements to this PR:
- [x] Add an e2e test
- [x] Don't print duplicate errors
- [x] Localize

### CODE REVIEW

File: src/vcpkg/portfileprovider.cpp [hunk 1]
```diff
@@ -172,18 +172,19 @@ namespace vcpkg::PortFileProvider
...
+                            auto scf_vspec = scf->get()->to_version_spec();
+                            if (scf_vspec == version_spec)
                             {
                                 return std::unique_ptr<SourceControlFileAndLocation>(
                                     new SourceControlFileAndLocation{std::move(*scf), std::move(*path)});
                             }
                             else
                             {
-                                return Strings::format("Error: Failed to load port from %s: names did "
+                                return Strings::format("Error: Failed to load port from %s: version specs did "
```
Comment (strega-nil-ms) [inline src/vcpkg/portfileprovider.cpp:183]: If you're gonna change this message anyways, I'd prefer this used the localized version

Comment (BillyONeal) [reply src/vcpkg/portfileprovider.cpp:183]: s/version/form/g :)

**Comment by BillyONeal** [comment]
I think investigating what happened was high priority but this has been in the product for more than a year, so I don't think avoiding adding the e2e test to get it a day or 2 earlier is reasonable.

As for not printing duplicates, isn't that a call to `Util::sort_unique_erase` away?

**Comment by BillyONeal** [comment]
> As for not printing duplicates, isn't that a call to `Util::sort_unique_erase` away?

It was, and I pushed that.

**Review by BillyONeal — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Localizing the error message changed the content which made the e2e test fail, I pushed a fix.

**Pull request merged**